snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/TrackerControllerImpl.kt

```diff
@@ -70,6 +71,42 @@ class TrackerControllerImpl  // Constructors
         return tracker.track(event)
     }
 
+
+    override fun decorateLink(uri: Uri, parameters: List<CrossDeviceParameter>): Uri? {
+        // UserId is a required parameter of `_sp`
+        if (this.session?.userId == null) {
```

Comment:
Should we do some logging here? Maybe a debug log that the userId is null and so the link couldn't be decorated? Something like we do here: https://github.com/snowplow/snowplow-android-tracker/blob/master/snowplow-tracker/src/main/java/com/snowplowanalytics/core/session/SessionControllerImpl.kt#L46C13-L46C16

Reply:
Added a log.